frontend/src/Crossword.tsx

```diff
@@ -384,29 +383,32 @@ const Crossword: React.FC<CrosswordProps> = (props) => {
     onExternalWrite.current = handleExternalWrite;
 
     const handleTouchStart = (event: TouchEvent) => {
-      if (event.touches.length > 1) {
-        // Start panning
+      if (event.targetTouches.length === 1) {
+        const [x, y] = getTouchLocation(event);
+        startDrawing(x, y);
+      } else {
         if (isDrawing) {
           stopDrawing();
         }
-        return;
       }
-      const [x, y] = getTouchLocation(event);
-      startDrawing(x, y);
     };
 
     const handleTouchMove = (event: TouchEvent) => {
-      if (event.touches.length > 1 || !isDrawing) {
-        // Panning
-        return;
+      if (event.targetTouches.length === 1 && isDrawing) {
+        const [x, y] = getTouchLocation(event);
+        draw(x, y);
+        // Firefox does not support the CSS touch-action: pinch-zoom property, 
+        // so we do preventDefault() instead
+        // @ts-ignore
+        if (typeof InstallTrigger !== 'undefined') { // Check if firefox v1.0 <
+          // console.log("Firefox!")
+          event.preventDefault();
+        }
       }
-      const [x, y] = getTouchLocation(event);
-      draw(x, y);
-      event.preventDefault();
     };
 
     const handleTouchEnd = (event: TouchEvent) => {
-      if (isDrawing && event.touches.length === 0) {
+      if (isDrawing && event.targetTouches.length === 0) {
```

Comment:
Is this the thing that solves the touch issue on your laptop or the `addEventListener` call?

Reply:
Nope, but it stops the drawing if you move the pointer/touch outside the canvas.